django-stubs/contrib/gis/gdal/raster/source.pyi

```diff
@@ -20,7 +22,7 @@ class TransformPoint(list[Sequence[float]]):
 
 class GDALRaster(GDALRasterBase):
     destructor: Any
-    def __init__(self, ds_input: Any, write: bool = ...) -> None: ...
+    def __init__(self, ds_input: str | Path | bytes | dict | c_void_p, write: bool = ...) -> None: ...
```

Comment:
I just noticed we have a shortcut for `str | Path`, but regardless, I have a question here: it’s unclear to me if the `c_void_p` type is allowed in the documentation or not. The django code tests `isinstance(…, c_void_p)` in the function, but I know it’s not the proper decision viewpoint.

See https://docs.djangoproject.com/en/4.2/ref/contrib/gis/gdal/#django.contrib.gis.gdal.GDALRaster

Reply:
I would keep `c_void_p` here sincethere's explicit support on Django side.